packages/@react-stately/table/src/useTableColumnResizeState.ts

```diff
@@ -0,0 +1,181 @@
+  const resizedColumnsRef = useRef<Set<Key>>(resizedColumns);
+
+  let getRealColumnWidth = useCallback((column: GridNode<T>) => {
+    let props = column.props as ColumnProps<T>;
+    return resizedColumns?.has(column.key) ? columnWidthsRef.current.get(column.key) : props.width ?? props.defaultWidth ?? getDefaultWidth?.(column.props) ?? '1fr';
+  }, [getDefaultWidth, resizedColumns]);
+
+  let getStaticAndDynamicColumns = useCallback((columns: GridNode<T>[]) : { staticColumns: GridNode<T>[], dynamicColumns: GridNode<T>[] } => columns.reduce((acc, column) => {
+    let width = getRealColumnWidth(column);
+    return isStatic(width) ? {...acc, staticColumns: [...acc.staticColumns, column]} : {...acc, dynamicColumns: [...acc.dynamicColumns, column]}; 
+  }, {staticColumns: [], dynamicColumns: []}), [getRealColumnWidth]);
+
+  let buildColumnWidths = useCallback((affectedColumns: GridNode<T>[], availableSpace: number): Map<Key, number> => {
+    const widths = new Map<Key, number>();
+    let remainingSpace = availableSpace;
+
+    const {staticColumns, dynamicColumns} = getStaticAndDynamicColumns(affectedColumns);
+
+    staticColumns.forEach(column => {
+      let width = getRealColumnWidth(column);
+      let w = parseStaticWidth(width, tableWidth.current);
+      widths.set(column.key, w);
+      remainingSpace -= w;
+    });
+
+    // dynamic columns
+    if (dynamicColumns.length > 0) {
+      const newColumnWidths = getDynamicColumnWidths(dynamicColumns, remainingSpace, tableWidth.current);
+      for (let column of newColumnWidths) {
+        widths.set(column.key, column.calculatedWidth);
+      }
+    }
+
+    return widths;
+  }, [getStaticAndDynamicColumns, getRealColumnWidth]);
+
+  // if the columns change, need to rebuild widths.
+  useLayoutEffect(() => {
+    columnsRef.current = columns;
+    const widths = buildColumnWidths(columns, tableWidth.current);
+    setColumnWidthsForRef(widths);
+  }, [columns, buildColumnWidths]);
+
+  function setColumnWidthsForRef(newWidths: Map<Key, number>) {
+    columnWidthsRef.current = newWidths;
+    // new map so that change detection is triggered
+    setColumnWidths(newWidths);
+  }
+
+  function setTableWidth(width: number) {
+    if (width && width !== tableWidth.current) {
+      tableWidth.current = width;
+      if (!isResizing.current) {
+        const widths = buildColumnWidths(columns, width);
+        setColumnWidthsForRef(widths);
+      }
+    }
+  }
+
+  function onColumnResizeStart() {
+    isResizing.current = true;
+    startResizeContentWidth.current = getContentWidth(columnWidthsRef.current);
+  }
+
+  function onColumnResize(column: GridNode<T>, width: number) {
+    let widthsObj = resizeColumn(column, width);
+    props.onColumnResize && props.onColumnResize(widthsObj);
+  }
+
+  function onColumnResizeEnd(column: GridNode<T>, width: number) {
+    isResizing.current = false;
+    if (props.onColumnResizeEnd) {
+      let widthsObj = resizeColumn(column, width);
```

Comment:
yes, i didn't specifically suggest adding a new ref for it because I was having trouble telling if another one was tracking something close to it or if you'd need a new one

Reply:
Made this change, I think it is a lot better thanks for the suggestion.